### PR TITLE
SLCORE-109 Simplify SonarLint Daemon Interface

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

File: daemon/src/main/java/org/sonarlint/daemon/model/ProxyLogOutput.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    if (response != null) {
+      LogEvent log = LogEvent.newBuilder()
+        .setLevel(level.name())
+        .setLog(formattedMessage)
+        .setIsDebug(level == Level.DEBUG || level == Level.TRACE)
+        .build();
+      try {
+        response.onNext(log);
+      } catch (StatusRuntimeException e) {
```
Comment (SonarTech) [inline daemon/src/main/java/org/sonarlint/daemon/model/ProxyLogOutput.java]: MAJOR Either log or rethrow this exception. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)

File: daemon/src/main/java/org/sonarlint/daemon/model/ProxyLogOutput.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        System.out.println("Log stream closed: " + e.getMessage());
```
Comment (SonarTech) [inline daemon/src/main/java/org/sonarlint/daemon/model/ProxyLogOutput.java]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS106)

File: daemon/src/main/java/org/sonarlint/daemon/model/ProxyLogOutput.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    if (this.response != null) {
+      this.response.onCompleted();
+    }
+    this.response = response;
+  }
+
+  @Override
+  public synchronized void log(String formattedMessage, Level level) {
+    if (level == Level.ERROR) {
+      System.err.println(formattedMessage);
```
Comment (SonarTech) [inline daemon/src/main/java/org/sonarlint/daemon/model/ProxyLogOutput.java]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS106)

File: daemon/src/main/java/org/sonarlint/daemon/model/ProxyLogOutput.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    } else {
+      System.out.println(formattedMessage);
```
Comment (SonarTech) [inline daemon/src/main/java/org/sonarlint/daemon/model/ProxyLogOutput.java]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS106)

File: daemon/src/main/java/org/sonarlint/daemon/model/ProxyIssueListener.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        severity = Severity.CRITICAL;
+        break;
+      case "MAJOR":
+      default:
+        severity = Severity.MAJOR;
+        break;
+    }
+
+    Issue.Builder builder = Issue.newBuilder();
+    builder.setRuleKey(issue.getRuleKey())
```
Comment (SonarTech) [inline daemon/src/main/java/org/sonarlint/daemon/model/ProxyIssueListener.java]: CRITICAL Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1067)

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 4 major

Watch the comments in this conversation to review them.

[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: daemon/src/main/java/org/sonarlint/daemon/model/ProxyLogOutput.java (hunk 1 above)
Comment (SonarTech) [inline daemon/src/main/java/org/sonarlint/daemon/model/ProxyLogOutput.java]: MAJOR Either log or rethrow this exception. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)

File: daemon/src/main/java/org/sonarlint/daemon/model/ProxyLogOutput.java (hunk 2 above)
Comment (SonarTech) [inline daemon/src/main/java/org/sonarlint/daemon/model/ProxyLogOutput.java]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS106)

File: daemon/src/main/java/org/sonarlint/daemon/model/ProxyLogOutput.java (hunk 3 above)
Comment (SonarTech) [inline daemon/src/main/java/org/sonarlint/daemon/model/ProxyLogOutput.java]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS106)

File: daemon/src/main/java/org/sonarlint/daemon/model/ProxyLogOutput.java (hunk 4 above)
Comment (SonarTech) [inline daemon/src/main/java/org/sonarlint/daemon/model/ProxyLogOutput.java]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS106)

File: daemon/src/main/java/org/sonarlint/daemon/model/ProxyIssueListener.java (hunk 5 above)
Comment (SonarTech) [inline daemon/src/main/java/org/sonarlint/daemon/model/ProxyIssueListener.java]: CRITICAL Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1067)

File: daemon/src/test/java/org/sonarlint/daemon/services/StandaloneSonarLintImplTest.java [hunk 6]
```diff
@@ -19,130 +19,29 @@
...
-import org.sonarsource.sonarlint.daemon.proto.SonarlintDaemon.Issue.Severity;
-import org.sonarsource.sonarlint.daemon.proto.SonarlintDaemon.LogEvent;
 import org.sonarsource.sonarlint.daemon.proto.SonarlintDaemon.StandaloneConfiguration;
 
-import io.grpc.StatusRuntimeException;
 import io.grpc.stub.StreamObserver;
 
 public class StandaloneSonarLintImplTest {
   @Rule
   public TemporaryFolder temp = new TemporaryFolder();
```
Comment (SonarTech) [inline daemon/src/test/java/org/sonarlint/daemon/services/StandaloneSonarLintImplTest.java]: MINOR Remove this unused "TemporaryFolder". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2924)

**Comment by SonarTech** [comment]
SonarQube analysis reported 6 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 4 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: daemon/src/main/assembly/dist-windows.xml [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+    <!-- sonarlint-daemon files: default config, embeded plugins, init script -->
+    <fileSet>
+      <directory>src/main/assembly/conf</directory>
+      <outputDirectory>conf</outputDirectory>
+      <fileMode>0644</fileMode>
+      <lineEnding>unix</lineEnding>
+    </fileSet>
+	
+	<fileSet>
```
Comment (janos-ss) [inline daemon/src/main/assembly/dist-windows.xml]: Overindented

File: daemon-protocol/src/main/proto/sonarlint-daemon.proto [hunk 8]
```diff
@@ -149,7 +149,6 @@ message ConnectedAnalysisReq {
 
 message StandaloneConfiguration {
   string home_path = 1;
```
Comment (janos-ss) [inline daemon-protocol/src/main/proto/sonarlint-daemon.proto]: Is this really necessary?

File: daemon/src/main/java/org/sonarlint/daemon/Utils.java [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+
+  @CheckForNull
+  private static Path getAppData() {
+    String osName = System.getProperty("os.name");
+
+    if (osName != null && osName.startsWith("Windows")) {
+      String appData = System.getenv("APPDATA");
+      if (appData != null) {
+        Path appDataPath = Paths.get(appData);
+        if (Files.exists(appDataPath)) {
```
Comment (SonarTech) [inline daemon/src/main/java/org/sonarlint/daemon/Utils.java]: MAJOR Replace this with a call to the "toFile().exists()" method [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3725)

File: daemon/src/main/java/org/sonarlint/daemon/Utils.java [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+    } else {
+      String userHome = System.getProperty("user.home");
+      home = Paths.get(userHome, "sonarlint");
+    }
+
+    try {
+      Files.createDirectories(home);
+
+      return home;
+    } catch (IOException e) {
```
Comment (SonarTech) [inline daemon/src/main/java/org/sonarlint/daemon/Utils.java]: MAJOR Either log or rethrow this exception. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)

**Comment by SonarTech** [comment]
SonarQube analysis reported 7 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 6 major

Watch the comments in this conversation to review them.

[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Comment by SonarTech** [comment]
SonarQube analysis reported 6 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 5 major

Watch the comments in this conversation to review them.

[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: daemon/src/main/java/org/sonarlint/daemon/model/ProxyLogOutput.java [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+
+import org.sonarlint.daemon.Logger;
+import org.sonarsource.sonarlint.core.client.api.common.LogOutput;
+import org.sonarsource.sonarlint.daemon.proto.SonarlintDaemon.LogEvent;
+
+import io.grpc.StatusRuntimeException;
+import io.grpc.stub.StreamObserver;
+
+public class ProxyLogOutput implements LogOutput {
+  private final Logger logger;
```
Comment (SonarTech) [inline daemon/src/main/java/org/sonarlint/daemon/model/ProxyLogOutput.java]: MAJOR Remove this unused "logger" private field. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1068)
MAJOR Make the "logger" logger private static final and rename it to comply with the format "LOG(?:GER)?". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1312)

File: daemon/src/main/java/org/sonarlint/daemon/services/ConnectedSonarLintImpl.java [hunk 12]
```diff
@@ -51,9 +51,14 @@
 import org.sonarsource.sonarlint.daemon.proto.SonarlintDaemon.Void;
 
 public class ConnectedSonarLintImpl extends ConnectedSonarLintGrpc.ConnectedSonarLintImplBase {
-  private static final Logger LOGGER = LoggerFactory.getLogger(ConnectedSonarLintImpl.class);
   private ConnectedSonarLintEngine engine;
-  private ProxyLogOutput logOutput = new ProxyLogOutput();
+  private final ProxyLogOutput logOutput;
+  private final Logger logger;
```
Comment (SonarTech) [inline daemon/src/main/java/org/sonarlint/daemon/services/ConnectedSonarLintImpl.java]: MAJOR Make the "logger" logger private static final and rename it to comply with the format "LOG(?:GER)?". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1312)

File: daemon/src/main/java/org/sonarlint/daemon/Daemon.java [hunk 13]
```diff
@@ -23,25 +23,27 @@
...
-import org.slf4j.LoggerFactory;
 import org.sonarlint.daemon.interceptors.ExceptionInterceptor;
 import org.sonarlint.daemon.services.ConnectedSonarLintImpl;
 import org.sonarlint.daemon.services.StandaloneSonarLintImpl;
 
 public class Daemon {
-  private static final Logger LOGGER = LoggerFactory.getLogger(Daemon.class);
   private static final int DEFAULT_PORT = 8050;
   private Server server;
+  private final Logger logger;
```
Comment (SonarTech) [inline daemon/src/main/java/org/sonarlint/daemon/Daemon.java]: MAJOR Make the "logger" logger private static final and rename it to comply with the format "LOG(?:GER)?". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1312)

File: daemon/src/main/java/org/sonarlint/daemon/Daemon.java [hunk 14]
```diff
@@ -23,25 +23,27 @@
...
+
+  Daemon(Logger logger) {
+    this.logger = logger;
+  }
 
   public static void main(String[] args) {
     int port;
-
+    Logger logger = new Logger();
```
Comment (SonarTech) [inline daemon/src/main/java/org/sonarlint/daemon/Daemon.java]: MAJOR Make the "logger" logger private static final and rename it to comply with the format "LOG(?:GER)?". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1312)

File: daemon/src/main/java/org/sonarlint/daemon/Logger.java [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+  public synchronized void info(String log) {
+    System.out.println(format(log, "INFO"));
+  }
+
+  private static String format(String log, String level) {
+    return String.format("%s [%s] %s %s", TIME_FORMAT.format(new Date()), Thread.currentThread().getName(), level, log);
+  }
+
+  public synchronized void error(String log, Exception e) {
+    System.err.println(format(log, "ERROR"));
```
Comment (SonarTech) [inline daemon/src/main/java/org/sonarlint/daemon/Logger.java]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS106)

File: daemon/src/main/java/org/sonarlint/daemon/Logger.java [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+    e.printStackTrace();
```
Comment (SonarTech) [inline daemon/src/main/java/org/sonarlint/daemon/Logger.java]: MINOR Use a logger to log this exception. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1148)

File: daemon/src/main/java/org/sonarlint/daemon/Logger.java [hunk 17]
```diff
@@ -0,0 +1,40 @@
...
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarlint.daemon;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class Logger {
+  private static final SimpleDateFormat TIME_FORMAT = new SimpleDateFormat("HH:mm:ss.SSS");
```
Comment (SonarTech) [inline daemon/src/main/java/org/sonarlint/daemon/Logger.java]: MAJOR Make "TIME_FORMAT" an instance variable. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2885)

File: daemon/src/main/java/org/sonarlint/daemon/Logger.java [hunk 18]
```diff
@@ -0,0 +1,40 @@
...
+
+  public synchronized void info(String log) {
+    System.out.println(format(log, "INFO"));
```
Comment (SonarTech) [inline daemon/src/main/java/org/sonarlint/daemon/Logger.java]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS106)

File: daemon/src/main/java/org/sonarlint/daemon/services/StandaloneSonarLintImpl.java [hunk 19]
```diff
@@ -19,198 +19,59 @@
...
 import org.sonarsource.sonarlint.daemon.proto.SonarlintDaemon.Void;
 import org.sonarsource.sonarlint.daemon.proto.StandaloneSonarLintGrpc;
 
+import io.grpc.stub.StreamObserver;
+
 public class StandaloneSonarLintImpl extends StandaloneSonarLintGrpc.StandaloneSonarLintImplBase {
-  private static final Logger LOGGER = LoggerFactory.getLogger(StandaloneSonarLintImpl.class);
+  private final ProxyLogOutput logOutput;
+  private final Collection<URL> analyzers;
+  private final Logger logger;
```
Comment (SonarTech) [inline daemon/src/main/java/org/sonarlint/daemon/services/StandaloneSonarLintImpl.java]: MAJOR Make the "logger" logger private static final and rename it to comply with the format "LOG(?:GER)?". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1312)

**Comment by SonarTech** [comment]
SonarQube analysis reported 16 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 14 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: daemon/src/main/java/org/sonarlint/daemon/Daemon.java [hunk 20]
```diff
@@ -23,25 +23,39 @@
...
+  Daemon(Logger logger) {
+    this.logger = logger;
+  }
 
   public static void main(String[] args) {
-    int port;
+    setUpNettyLogging();
 
+    int port;
+    Logger logger = new Logger();
```
Comment (SonarTech) [inline daemon/src/main/java/org/sonarlint/daemon/Daemon.java]: MAJOR Make the "logger" logger private static final and rename it to comply with the format "LOG(?:GER)?". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1312)

**Pull request merged**